src/python/deepgnn/graph_engine/snark/decoders.py

```diff
@@ -43,6 +43,225 @@ def decode(self, line: str) -> Iterator[Tuple[int, int, int, float, list]]:
 DecoderType = TypeVar("DecoderType", bound=Decoder)
 
 
+class EdgeListDecoder(Decoder):
+    r"""Convert the text line into node object.
+
+    Edge List Format:
+        ```
+        <node_0_info>
+        <edge_0,1_info>
+        <edge_0,2_info>
+        ...
+        <node_1_info>
+        ...
+        ```
+        node_info: node_id,-1,node_type,node_weight,<features>
+        edge_info: src,edge_type,dst,edge_weight,<features>
+
+        Sort the file so the first line has the first node's info, the next few lines have all the first node's
+        outgoing edges. Then the next line will have the second node's info and so on.
+
+        Feature fectors to fill <features> can be dense or sparse. Features will be given
+        indexes starting at 0 and indexes can be skipped with 0 length vectors. Each node and
+        edge does not have to have the same number of features or feature types.
+
+        features[dense]: dtype_name,length,v1,v2,...,dtype_name2,length2,v1,v2,...
+        features[sparse with 2 dim coordinates vector]: dtype_name,values.size/coords.shape[1],c1,c2,...,v1,v2,...
```

Comment:
```python        
        features[sparse with 2 dim coordinates vector]: dtype_name,values.size/coords.shape[1],c1,c2,...,v1,v2,...
        features[sparse with 1 dim coordinates vector]: dtype_name,values.size/0,c1,c2,...,v1,v2,...        
```

1. It is not clear what coords.shape[1] is and why do we use second dimension, not first.
2. Why there is a difference between 2 and 1 dimensional vectors? Can we just keep the same number?
3. How do we differentiate between multidimensional coordinate items? Does `c1,c2,..` mean coordinates for values `v1, v2` or coordinates for value `v1` for the first 2 dimensions?

Reply:
I've updated these.